src/kktsystem.jl

```diff
@@ -0,0 +1,339 @@
+
+abstract type AbstractKKTSystem{T, MT} end
```

Comment:
I am not 100% sure about that name. I am wondering whether `AbstractKKTLinearSystem` would be more explicit?

Reply:
Good catch!